### PR TITLE
feat: add incidents search endpoint structure

### PR DESCRIPTION
**Pull request by ralfpuchert**
## Description

* Add Incident endpoint structure for /v2/incidents/search
* Add Controller, Filter, Sort components
* Tests will follow in https://github.com/camunda/camunda/pull/21415

## Checklist

- [ ] for CI changes:
  - [ ] structural/foundational changes signed off by [CI DRI](https://github.com/cmur2)
  - [ ] [ci.yml](https://github.com/camunda/camunda/blob/main/.github/workflows/ci.yml) modifications comply with ["Unified CI" requirements](https://github.com/camunda/camunda/wiki/CI-&-Automation#workflow-inclusion-criteria)

## Related issues

Related to #20661

### CODE REVIEW

File: zeebe/gateway-rest/src/main/java/io/camunda/zeebe/gateway/rest/controller/IncidentQueryController.java [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+        .fold(RestErrorMapper::mapProblemToResponse, this::search);
+  }
+
+  private ResponseEntity<IncidentSearchQueryResponse> search(final IncidentQuery query) {
+    try {
+      final var result =
+          incidentServices.withAuthentication(RequestMapper.getAuthentication()).search(query);
+      return ResponseEntity.ok(SearchQueryResponseMapper.toIncidentSearchQueryResponse(result));
+    } catch (final Throwable e) {
+      final var problemDetail =
```
Comment (nathansandi) [inline zeebe/gateway-rest/src/main/java/io/camunda/zeebe/gateway/rest/controller/IncidentQueryController.java:51]: 🔧  Related to my previous comment, 
This error could be managed like this 

```
  } catch (final ValidationException e) {
      final var problemDetail =
          RestErrorMapper.createProblemDetail(
              HttpStatus.BAD_REQUEST,
              e.getMessage(),
              "Validation failed for UserTask Search Query");
      return RestErrorMapper.mapProblemToResponse(problemDetail);
    } catch (final Exception e) {
      final var problemDetail =
          RestErrorMapper.createProblemDetail(
              HttpStatus.INTERNAL_SERVER_ERROR,
              e.getMessage(),
              "Failed to execute UserTask Search Query");
      return RestErrorMapper.mapProblemToResponse(problemDetail);
```

So we separate validation erros vs. Internal ones

**Review by nathansandi — CHANGES_REQUESTED** [review]
Great work here, just minors suggestions/improvements.

File: service/src/main/java/io/camunda/service/search/filter/IncidentFilter.java [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+import java.util.function.Function;
+
+public record IncidentFilter(
+    List<Long> keys,
+    List<Long> processDefinitionKeys,
+    List<Long> processInstanceKeys,
+    List<String> types,
+    List<String> flowNodeIds,
+    List<String> flowNodeInstanceIds,
+    DateValueFilter creationTimeFilter,
```
[the remark this answers is not in the document]
Comment (ralfpuchert) [reply service/src/main/java/io/camunda/service/search/filter/IncidentFilter.java]: Thank you, haven't noticed yet. Will remove this.

**Review by nathansandi — APPROVED** [review]
Great work 💯  :)

**Pull request merged**